src/server/routes/collection.js

```diff
@@ -85,4 +84,28 @@ router.get('/:collectionId', (req, res) => {
 	res.status(200).send(collection);
 });
 
+/* eslint-disable no-await-in-loop */
+router.post('/:collectionId/add', auth.isAuthenticated, auth.isCollectionOwnerOrCollaborator, async (req, res) => {
+	const {bbids} = req.body;
+	const {collection} = res.locals;
+	try {
+		const {UserCollectionItem} = req.app.locals.orm;
+		for (const bbid of bbids) {
+			try {
+				await new UserCollectionItem({
+					bbid,
+					collectionId: collection.id
+				}).save(null, {method: 'insert'});
+			}
+			catch (err) {
+				// entity already in the collection
```

Comment:
Umm I don't really think we need to log this. Because user can accidentally add same entity multiple times and it's not really a bug. 
It would've been nice if bookshelf had `upsert` option.
The other thing we can do is first check if exists, if not then add it. But that would require 2 calls. And since this is gonna be used alot, I don't think we want this.

Reply:
Yes, upsert would be useful here, definitely.
Let's not do the double call, as therse are no dire consequences if adding an entity twice.
The logging I meant more in case there is another type of error.

This can be achieved by checking the type of error, displaying an "entity already added" message if it's a [NoRowsUpdatedError](https://bookshelfjs.org/api.html#Model-static-NoRowsUpdatedError) and logging the error otherwise.